### ISSUE TITLE
Something wrong with the vim colorshemes

### ISSUE THREAD

**Issue by aeilot** [microsoft/terminal#6755]
# Environment

```none
Windows build number:  10.0.19042.0
Windows Terminal version (if applicable): 1.0.1811.0
VIM: 8.2
```

# Steps to reproduce

I installed [onedark.vim](https://github.com/joshdick/onedark.vim) colorscheme for vim.

But the colorscheme doesn't work correctly as I saw on my Mac.

By the way, I am using the VSCode colorscheme `snazzy`.

Here's it's config:
```json
		{
				"name": "Snazzy",
				"black": "#000000",
				"red": "#fc4346",
				"green": "#50fb7c",
				"yellow": "#f0fb8c",
				"blue": "#49baff",
				"purple": "#fc4cb4",
				"cyan": "#8be9fe",
				"white": "#ededec",
				"brightBlack": "#555555",
				"brightRed": "#fc4346",
				"brightGreen": "#50fb7c",
				"brightYellow": "#f0fb8c",
				"brightBlue": "#49baff",
				"brightPurple": "#fc4cb4",
				"brightCyan": "#8be9fe",
				"brightWhite": "#ededec",
				"background": "#1e1f29",
				"foreground": "#ebece6"
		}
```

```json
	        "acrylicOpacity": 0.7,
                "colorScheme" : "Snazzy",
                "cursorColor" : "#000000",
                "fontFace" : "MesloLGL Nerd Font",
                "useAcrylic": true
```

# Expected behavior

This is what `onedark.vim` really looks like:
![](https://raw.githubusercontent.com/joshdick/onedark.vim/master/img/readme_header.png)

# Actual behavior

But it looks like this here:
![image](https://user-images.githubusercontent.com/25583228/86315265-49d60f80-bc5c-11ea-83a9-532e27284362.png)

**Comment by aeilot** [comment]
As I checked, it's the `useAcrylic` and the opacity's fault. Once I disable these features, the vim will be just like what I expected.

**Comment by DHowett** [comment]
This is a version of #293 that is not fixed by #6698 #6506. _vim.exe_ achieves setting this "background" by _explicitly redefining what the console color at index 0 means_. This is actually _worse_ after #6506 because we're better about exposing the drawing operations of the application to the terminal, and we still believe console color 0 to be the "background". When an application uses the legacy console API to set the colors, we drop the background and foreground out so the acrylic can shine through.

This is very dangerously close to by design and might need a fix in vim.exe.

**Comment by gordonmessmer** [comment]
Is there a preview build that includes 6698 and 6506?

**Comment by zadjii-msft** [comment]
@gordonmessmer at this point, the 1.2 preview should be out that includes those changes.

**Comment by zadjii-msft** [comment]
I'm thinking that we might be able to fix this if we had conpty listen for changes to the color table, and emit those with the OSC sequences for setting the color table. I don't think this is the _right_ solution - `vim.exe` could definitely just fix this on their end. But this is an option that's available to us

**Comment by aeilot** [comment]
Thanks for replying!

On Linux with KDE, the Yakuake Terminal which also contains the feature of `acrylic`, it appears like this:

![Screenshot_20200805_115116](https://user-images.githubusercontent.com/25583228/89369913-fd299c80-d711-11ea-8695-f53827c99207.png)

![Screenshot_20200805_114830](https://user-images.githubusercontent.com/25583228/89369852-d53a3900-d711-11ea-849a-d64bf7d23fbf.png)

On Mac, it looks like this as well.

**Maybe we can do as they do.** What I think about it is, when the user opens special apps like vim, we can disable this feature or etc.

**Comment by DHowett** [comment]
This is because on those platforms Vim uses VT sequences, but on Windows it does not. If you use  Vim inside WSL you can see that it will work the same way :smile:

**Comment by j4james** [comment]
> I'm thinking that we might be able to fix this if we had conpty listen for changes to the color table, and emit those with the OSC sequences for setting the color table.

I don't think this will help. I wrote a patch to fix issue #2985 that does something like this, and it makes no difference to vim. The underlying problem is that the console API can't differentiate between a black background and default, so we treat them as the same thing (that is they're both interpreted as default). So it doesn't matter what the palette color for black is set to - a black background is still going to be interpreted as default, and thus will always be transparent (if you have acrylic enabled).
